salvo/src/lib/test_source_manager.py

```diff
@@ -40,23 +77,57 @@ def test_get_envoy_images_for_benchmark(mock_run_command):
   """Verify that we can determine the current and previous image
      tags from a minimal job control object.
   """
-  pass
 
-def _run_command_side_effect_for_disk_files(*args):
-  """Adjust the check_call output so that we can respond differently to
-     input arguments.
+  job_control = proto_control.JobControl(
+      remote=False,
+      scavenging_benchmark=True
+  )
 
-  Args:
-    args: the list of arguments received by the mocked function
-  """
-  pass
+  image_config = _generate_default_benchmark_images(job_control)
+  image_config.envoy_image = "envoyproxy/envoy-dev:latest"
+
+  mock_run_command.side_effect = _run_command_side_effect
+
+  src_mgr = source_manager.SourceManager(job_control)
+  hashes = src_mgr.get_envoy_hashes_for_benchmark()
+
+  assert hashes == ['mocked_hash_the_sequel', 'latest']
 
 @mock.patch("src.lib.cmd_exec.run_command")
 def test_previous_hash_with_disk_files(mock_run_command):
```

Comment:
Similarly, this method should probably be called `test_get_image_hashes_from_disk_source`.

Using names that match the tested methods helps future contributors to find existing tests for a method they want to modify.

Reply:
Makes sense.  I can update.